File: 1-two-sum/cpp/main.cpp
class Solution {
 public:
  vector<int> twoSum(vector<int> &nums, int target) {
    sort(nums.begin(), nums.end());
    int l = 0;
    int r = nums.size() - 1;

    while (l < r) {
      int sum = nums[l] + nums[r];

      if (sum < target) {
        ++l;
      } else if (sum > target) {
        --r;
      } else {
        return {l, r};
      }
    }

    return {0, 0};
  }
};

File: 1-two-sum/python/main.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s = {}
        for i, n in enumerate(nums):
            if target - n in s:
                return [s[target - n], i]
            else:
                s[n] = i
        return [-1, -1]
